### DISCUSSION TITLE
Three Network material model PolyUMod

### DISCUSSION THREAD

**Discussion by tstepan333** [idaholab/moose#20245] [Q&A Modules: Solid mechanics] [answered]
Hello,

Is it possible to import [Three Network material model from PolyUMod](https://polymerfem.com/three-network-model/) to Moose? Or can I implement the Three Network model directly in Moose? 

Thanks,
Tatiana

**Comment by GiudGiud** [comment, marked as answer]
@jiangwen84 @hugary1995  do you know about this PolyUMod?

**Reply by jiangwen84** [reply]
I doubt it can be directly imported to MOOSE, but you should be able to implement in MOOSE's TensorMechanics module. You can consider to use this new Lagrangian Mechanics staff, see https://github.com/idaholab/moose/pull/17475
